### DISCUSSION TITLE
How to disable nextjs's random locale redirects?

### DISCUSSION THREAD

**Discussion by CaptainN** [vercel/next.js#37759] [Help]
I have a multi-locale site, and if I just use the standard default route (en in my case) with links that have a /en/ prefix, everything works great - at runtime. Like, clicking between the pages works, and I stay in the /en/ prefix for every link. But for some reason, if the user loads a link that has /en/ in it, it forwards to the non-prefixed version of the same link after it loads. 

Why does it do that? And why does it only do that on a fresh load? It definitely does NOT do that when you navigate to the same page.

Anyway, I just want to disable that. Is there any way?

**Comment by vikingair** [comment]
In the Next config you can have this:

```ts
i18n: {
  locales: ['default', 'en'],
  defaultLocale: 'default',
},
```

The `defaultLocale` is the one that is used when your app doesn't show any language in the pathname. If you use instead `"en"` as the default, you run into the above mentioned behavior.

But the problem is that if the user would now call your app without the "/en/" in the path **and** doesn't list "en" in the Browser languages, then NextJS will use the value "default" as locale.

To prevent that, you will need to configure an additional static redirect for those cases or have a middleware that doesn't allow the "default" locale. But beware, all files served from the "public" directory or served under "/_next/" need to be ignored and will use the "default" locale. Don't know why, but these requests sometimes are also processed by the  `_document.tsx` and `_app.tsx` files.

Sorry, that there is no easy answer to it. Would have also loved that the "defaultLanguage" wasn't designed that way.
